mautic/mautic
# Show "added through API" event log to timeline

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | staging
| Bug fix?                               | no
| New feature?                           | yes
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | TODO
| Related developer documentation PR URL | TODO
| Issue(s) addressed                     | N/A




#### Description:

~~⚠️ **NEEDS #7538 TO BE MERGED FIRST**~~ DONE ✅ 

Builds upon #7538. That PR adds entries to the `lead_event_log` table when a contact is created through the API. This PR shows those entries in the UI, so that the user can see that the contact was created with the API.



#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Go to https://reqbin.com/
3. Use this config (url = https://mautibox.com/9067/api/contacts/new, credentials: admin / mautic):
![image](https://user-images.githubusercontent.com/17739158/90336856-d0f9ff80-dfde-11ea-8c77-3761df145876.png)
4. In the "content" tab, copy and paste this data:

```JSON
{
      "email": "demo@demo.com",
      "firstname": "testing",
      "lastname": "test"
}
```
5. Click "Send"
6. Go to contacts: https://mautibox.com/9067/s/contacts
7. When the contact is created, you'll now see an event "Added through API":

![image](https://user-images.githubusercontent.com/17739158/89740609-87635f00-da8a-11ea-852e-8e6316c23937.png)
